File: 1913-maximum-product-difference-between-two-pairs/1913-maximum-product-difference-between-two-pairs.java
class Solution {
    public int maxProductDifference(int[] nums) {
        int big = 0, sBig = 0;
        int small = Integer.MAX_VALUE, sSmall = Integer.MAX_VALUE;
        for(int x : nums){
            if(x > big){
                sBig = big;
                big = x;
            }
            else sBig = Math.max(sBig, x);
            if(x < small){
                sSmall = small;
                small = x;
            }
            else sSmall = Math.min(sSmall, x);
        }
        return big*sBig - small*sSmall;
    }
}

File: 1913-maximum-product-difference-between-two-pairs/1913-maximum-product-difference-between-two-pairs.cpp
class Solution:
    def maxProductDifference(self, nums: List[int]) -> int:
        big, sBig, small, sSmall = 0, 0, float('inf'), float('inf')
        for x in nums:
            if x > big:
                sBig = big
                big = x
            else: sBig = max(sBig, x)
            if x < small:
                sSmall = small
                small = x
            else: sSmall = min(sSmall, x)
        return big*sBig - small*sSmall